src/libPersistence/BlockStorage.cpp

```diff
@@ -455,6 +532,8 @@ bool BlockStorage::GetAllDSBlocks(std::list<DSBlockSharedPtr>& blocks) {
 bool BlockStorage::GetAllTxBlocks(std::list<TxBlockSharedPtr>& blocks) {
   LOG_MARKER();
 
+  shared_lock<shared_timed_mutex> g(m_mutexTxBlockchain);
+
   leveldb::Iterator* it =
       m_txBlockchainDB->GetDB()->NewIterator(leveldb::ReadOptions());
```

Comment:
Maybe let `m_mutexTxBlockchain` to protect `leveldb::Iterator* it = m_txBlockchainDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop